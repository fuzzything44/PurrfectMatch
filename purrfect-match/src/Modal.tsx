import React from 'react';

interface ModalProps {
    display: boolean;
    onClose: () => void
};

interface ModalState {

};

export class Modal extends React.Component<ModalProps, ModalState> {

    public render() {
        if (!this.props.display) {
            return null;
        }
        return (<div className="modal-overlay" onClick={() => this.props.onClose()}>
            <div className="modal" onClick={(e) => e.stopPropagation() }>
                <button className="modal-close" onClick={() => this.props.onClose()} >X</button>
                {this.props.children}
            </div>
        </div>);
    }

}


