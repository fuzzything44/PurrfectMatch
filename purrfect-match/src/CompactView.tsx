import React from 'react';
import { Modal } from "./Modal";
import { Redirect } from "react-router-dom";
import cat from "./pictures/cat5.jpg";

interface compactViewProps {
    name: string;
    breed: string;
    age: string;
    gender: string;
};

interface compactViewState {
    modalOpen: boolean;
};

export class CompactView extends React.Component<compactViewProps, compactViewState> {
   
    constructor(props: any) {
        super(props);

        this.state = {
            modalOpen: false
        };
    }

    public render() {
        return (
            <div>
            <button className="compactView" onClick={() => { this.setState({ modalOpen: true }) }}>
                <img id="animalImage" src={cat} alt="Image of animal"/>
                <h1 id="animalName">
                    {this.props.name}
                </h1>
                <p id="animalInfo">
                    Breed: {this.props.breed}
                    <br/> Age: {this.props.age} 
                    <br/> Gender: {this.props.gender}
                </p>
            </button>

            <Modal display={this.state.modalOpen} onClose={() => this.setState({ modalOpen: false })}>
            <img id="successStoryPageImage" src={cat} alt="Cat"/> 
            <div id="successStoryPageText">
                <h1 id="successStoryPageHeader"> 
                    Name: Ollie
                    <br />Age: 3 years
                    <br />Breed: American Shorthair
                    <br />Adopted: 10/24/2019
                    <br />
                    <br />
                    A message from Ollie's owner, Bob:
                </h1>
                <p> 
                    Ollie has been a real friend! Coming home from a crazy day at work, I get to look forward to this goofball greeting me with lots of purring and cuddles. I found Ollie through Purrfect Match. I couldn’t go to my local shelter without first knowing if I had a specific pet in mind, because I can’t be away from work for very long during business hours. Perrfect Match helped me find Ollie, and I’m so happy he’s in my life! He’s a cutie, and I can’t imagine not having found him.
                </p>
            </div>
            </Modal>
            
            </div>
        )
    }

}
