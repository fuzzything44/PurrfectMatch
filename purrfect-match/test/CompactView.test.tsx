import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CompactView } from 'CompactView';
import { Modal } from 'Modal';

configure({ adapter: new Adapter() });

jest.mock("Modal");

let wrapper: ShallowWrapper<any, any, any>;
beforeEach(() => {
    wrapper = shallow(<CompactView name="mockName" breed="someBreed" age="mockAge" gender="mockGender" />);
});

test("Snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
    expect("Test failure: we still need to populate the modal. ").toBeFalsy();
});

test("Modal opens when clicked", () => {
    // This needs to be split like this due to inconsistent type definitions with enzyme. 
    // When using wrapper.find(Modal) it knows it'l get a Modal so can fill in the props type
    // correctly. However, since button is a react component we need to search by string, leaving
    // props empty without this cast;
    let button = wrapper.find("button") as any;
    button.props().onClick();

    expect(wrapper.find(Modal).props().display).toBeTruthy();
});

test("Modal closes when closed", () => {
    wrapper.setState({modalOpen: true});

    expect(wrapper.find(Modal).props().display).toBeTruthy();

    wrapper.find(Modal).props().onClose();

    expect(wrapper.find(Modal).props().display).toBeFalsy();
});
