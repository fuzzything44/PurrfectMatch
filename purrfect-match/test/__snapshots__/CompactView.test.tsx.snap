// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div>
  <button
    className="compactView"
    onClick={[Function]}
  >
    <img
      alt="Image of animal"
      id="animalImage"
      src="test-file-stub"
    />
    <h1
      id="animalName"
    >
      mockName
    </h1>
    <p
      id="animalInfo"
    >
      Breed: 
      someBreed
      <br />
       Age: 
      mockAge
      <br />
       Gender: 
      mockGender
    </p>
  </button>
  <Modal
    display={false}
    onClose={[Function]}
  >
    <img
      alt="Cat"
      id="successStoryPageImage"
      src="test-file-stub"
    />
    <div
      id="successStoryPageText"
    >
      <h1
        id="successStoryPageHeader"
      >
        Name: Ollie
        <br />
        Age: 3 years
        <br />
        Breed: American Shorthair
        <br />
        Adopted: 10/24/2019
        <br />
        <br />
        A message from Ollie's owner, Bob:
      </h1>
      <p>
        Ollie has been a real friend! Coming home from a crazy day at work, I get to look forward to this goofball greeting me with lots of purring and cuddles. I found Olliethrough    Purrfect Match. I couldn’t go to my local shelter without first knowing if I had a specific pet in mind, because I can’t be away from work for very longduring business          hours. Perrfect Match helped me find Ollie, and I’m so happy he’s in my life! He’s a cutie, and I can’t imagine not having found him.
      </p>
    </div>
  </Modal>
</div>
`;
